src/main/java/dev/j3fftw/litexpansion/machine/MassFabricator.java

```diff
@@ -58,15 +55,15 @@ public MassFabricator() {
 
     private void setupInv() {
         createPreset(this, "&5Mass Fabricator", blockMenuPreset -> {
-            for (int i : BORDER) {
+            for (int i = 0; i < 27; i++)
                 blockMenuPreset.addItem(i, ChestMenuUtils.getBackground(), ChestMenuUtils.getEmptyClickHandler());
-            }
+
+            for (int slot : INPUT_SLOTS)
+                blockMenuPreset.addItem(slot, null, (player, i, itemStack, clickAction) -> true);
```

Comment:
instead of this, you can just do

`if (i == INPUT_SLOTS[0] || i == INPUT_SLOTS[1]) continue;` in the for loop above